src/docs/product/session-replay/performance-overhead.mdx

```diff
@@ -24,9 +24,36 @@ Sentry's Session Replay SDK takes several measures to avoid negatively impacting
+
+### Pageload stage
+
+- [SDK bundle size is something we optimize for.](https://blog.sentry.io/sentry-bundle-size-how-we-reduced-replay-sdk-by-35/) Make sure you have tree-shaking enabled, and have updated to the latest SDK version (at least version 7.78.0) for the smallest bundle sizes.
+- The size of your initial HTML page affects how long the SDK takes to read the initial snapshot. Using your browser devtools, filter the Network panel to show "type=document" and look at the size to get a sense of the impact on your site. A sterotypical single-page-app that shows a loading screen and needs to wait for JavaScript to render everything will have negligible overhead from the initial snapshot.
+- All Replay SDK data is compressed before being sent across the network. HTML is highly compressable, we were surprised how effective compression is on real websites. Network overhead is further reduced if you have [PII Masking and Blocking](https://docs.sentry.io/platforms/javascript/session-replay/privacy/) enabled (which is the default). Masked text is replaced with `*` which gzip shrinks really well, while blocked content is not included at all.
+
+### Capturing mouse/touch events
+
+- Mouse-move, click/touch & hover events are incredibly small and fast to log; overhead is negligible.
+- We deduplicate click events (while still detecting rage clicks) and use smoothing for mouse-move events to balance the detail you see while debugging, with the amount of redundant data captured.
+
+### Dynamic content & DOM mutations
+
+- The frequency of DOM mutations is highly application-specific. If there are no mutations happening then you will not see any replay data transmitted from the SDK to Sentry. On the other hand, if you have JavaScript timers that update the HTML on a regular basis, you will see a steady stream of HTML changes sent on the network.
+- The size of DOM mutations is also highly application-specific. Adding/removing/changing more DOM nodes at once will increase the time it takes to parse and minify the data. Keep in mind though that the overhead here happens after changes are made to the DOM as they are shown to the user (inside a `MutationObserver` callback).
+  The SDK has [mutation limits](https://docs.sentry.io/platforms/javascript/session-replay/configuration/#mutation-limits) to mitigate a negative user experience: changing 750 nodes will emit a warning breadcrumb into the replay, while emitting 10,000+ nodes will cause the replay to end.
+
+### Page unload stage
+
+- The SDK has no special handling for page unload; anything in memory will be deallocated by the browser and the user will transition to their next web page as quickly as possible.
+
+## Measuring Performance Overhead
+
+The only way to get accurate metrics is to measure performance overhead yourself. You can apply realistic access patterns against your own website and infrastructure, and correlate that to your own topline business metrics.
```

Comment:
it'd be cool to have a self instrumentation mode, sentry perf for the replay sdk

Reply:
ya, like to show sdk spans and stuff? that'd be fun. 

I think also the product is getting better at exposing these different performance concerns. not there yet though because it's harder than just counting bytes unfortunately.